pandas/tests/io/test_feather.py

```diff
@@ -74,6 +77,18 @@ def test_stringify_columns(self):
         df = pd.DataFrame(np.arange(12).reshape(4, 3)).copy()
         self.check_error_on_write(df, ValueError)
 
+    def test_read_columns(self):
+
+        df = pd.DataFrame({'col1': list('abc'),
+                           'col2': list(range(1, 4)),
+                           'col3': list('xyz'),
+                           'col4': list(range(4, 7))})
+        self.check_round_trip(df, columns=None)
+        self.check_round_trip(df, columns=df.columns)
+        cols = ['col3', 'col1']
+        self.check_round_trip(df, expected=df[cols],
+                              columns=cols)
```

Comment:
Anyway we can use `pytest.parametrize` on this?

Reply:
Is there even a need for the first test passing `df.columns`? I think we just need to ensure that `columns` is passed through to `read_feather`. Anything else to do with the types / contents of `columns` is between the user and pyarrow.

Reply:
> Anyway we can use `pytest.mark.parametrize` on this (with respect to `columns`)?

Some thing like this ?
```
    @pytest.mark.parametrize("columns", [
        None,
        ['col3', 'col1']
    ])
    def test_read_columns(self, columns):
        # GH 24025
        df = pd.DataFrame({'col1': list('abc'),
                           'col2': list(range(1, 4)),
                           'col3': list('xyz'),
                           'col4': list(range(4, 7))})
        self.check_round_trip(df, expected=df[columns or df.columns],
                              columns=columns)
```

Reply:
I wouldn't do `columns or df.columns`.  Just pass in `None`, `['col1', 'col2', 'col3', 'col4']`, and `['col3', 'col1']` as three possible tests.

Reply:
> I wouldn't do `columns or df.columns`. Just pass in `None`, `['col1', 'col2', 'col3', 'col4']`, and `['col3', 'col1']` as three possible tests.

I too hate to use the `or` statement but when passing `None` -> `df[None]` will throw key error,

Reply:
Right, but you can condition the value of `expected` depending on what `columns` is.  Instead of "hard-coding" (for lack of a better term) to `df[<indexer>]`, you can set it to be `df` is `"columns"` is `None` and then index otherwise.